zcash_primitives/src/transaction/builder.rs

```diff
@@ -152,7 +157,7 @@ impl Progress {
 }
 
 /// Generates a [`Transaction`] from its inputs and outputs.
-pub struct Builder<'a, P, R> {
+pub struct Builder<'a, P, R, U: sapling::builder::ProverProgress> {
```

Comment:
I presume that when/if we implement a progress notifier for Orchard, that we'd bound on both those traits?

One question, how does this end up working if we feature-flag Sapling off?

Reply:
That's gonna start getting messy very quickly, which is partly why I did not go down the trait pathway to begin with, and only did so at your request.

It's probably doable by replacing this bound with a top-level insulatory bound, and then having an intermediate top-level struct that wraps the Sapling and Orchard type impls and applies relevant feature-flagging.